### ISSUE TITLE
all regions get count:0,but avg loss still decrease

### ISSUE THREAD

**Issue by jacobssy** [AlexeyAB/darknet#3831]
hi~ i met a problem that all regions get count:0 which means nan,however the avg loss still decrease ,as the log shows:
(next mAP calculation at 4000 iterations)                                                                                                            │·····························
 624: 10.911251, 24.309635 avg loss, 0.000001 rate, 5.879649 seconds, 159744 images                                                                   │·····························
Saving weights to /data/home/sunsiyuan/tengdun/darknet_aug/darknet-master/backup//yolov3_last.weights                                                 │·····························
Loaded: 0.000049 seconds                                                                                                                              │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 82 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.104351, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 82 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.105609, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 94 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.044726, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 82 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.108288, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 94 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.045016, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 94 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.046134, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 106 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.018187, .5R: -nan, │·····························
.75R: -nan, count: 0                                                                                                                                  │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 106 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.018389, .5R: -nan, │·····························
.75R: -nan, count: 0                                                                                                                                  │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 106 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.018654, .5R: -nan, │·····························
.75R: -nan, count: 0                                                                                                                                  │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 82 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.108125, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 94 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.046022, .5R: -nan, .│·····························
75R: -nan, count: 0                                                                                                                                   │·····························
v3 (mse loss, Normalizer: (iou: 0.750000, cls: 1.000000) Region 106 Avg (IOU: -nan, GIOU: -nan), Class: -nan, Obj: -nan, No Obj: 0.018496, .5R: -nan, │·····························
.75R: -nan, count: 0

**Comment by AlexeyAB** [comment]
It seems all your labels are empty.

Did you recalculate your anchors? - Don't do it.

**Comment by jacobssy** [comment]
@AlexeyAB thanks for your reply, I did not recalculate anchors. i just modify the hyper parameters by your guide,like lr,max_iters...

**Comment by jacobssy** [comment]
@AlexeyAB now the loss is 0.. Lol, but all count still 0, is there any possible that my data format is incorrect?it can be trained successfully in the  pjreddie/darknet yolo3.cfg,and the result is also normal.

**Comment by AlexeyAB** [comment]
Train with flag ` -show_imgs` it should show bounded boxes on your training images, are they correct?

**Comment by jacobssy** [comment]
@AlexeyAB it seems no bounded boxes on training images...

**Comment by AlexeyAB** [comment]
Your training dataset is wrong.

Check it by using  https://github.com/AlexeyAB/Yolo_mark

**Comment by jacobssy** [comment]
i checked,my fault. dataset is not correct
